pkg/util/azureclient/mgmt/compute/virtualmachinescalesetvms_addons.go

```diff
@@ -0,0 +1,41 @@
+package compute
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+)
+
+type VirtualMachineScaleSetVMsClientAddons interface {
+	ListVMs(ctx context.Context, resourceGroupName string, virtualMachineScaleSetName string, filter string, selectParameter string, expand string) ([]compute.VirtualMachineScaleSetVM, error)
+	RunCommandAndWait(ctx context.Context, resourceGroupName string, VMScaleSetName string, instanceID string, parameters compute.RunCommandInput) error
+}
+
+func (c *virtualMachineScaleSetVMsClient) RunCommandAndWait(ctx context.Context, resourceGroupName string, VMScaleSetName string, instanceID string, parameters compute.RunCommandInput) error {
+	future, err := c.VirtualMachineScaleSetVMsClient.RunCommand(ctx, resourceGroupName, VMScaleSetName, instanceID, parameters)
+	if err != nil {
+		return err
+	}
+
+	return future.WaitForCompletionRef(ctx, c.VirtualMachineScaleSetVMsClient.Client)
+}
+
+func (c *virtualMachineScaleSetVMsClient) ListVMs(ctx context.Context, resourceGroupName string, virtualMachineScaleSetName string, filter string, selectParameter string, expand string) ([]compute.VirtualMachineScaleSetVM, error) {
+	var scaleSetsVMs []compute.VirtualMachineScaleSetVM
+	result, err := c.VirtualMachineScaleSetVMsClient.List(ctx, resourceGroupName, virtualMachineScaleSetName, filter, selectParameter, expand)
+	if err != nil {
+		return nil, err
+	}
+	scaleSetsVMs = append(scaleSetsVMs, result.Values()...)
+	for result.NotDone() {
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
are you absolutely sure this comes above reading result.Values()?

Reply:
I think this this all together created right behviour (?)
1. List
2. Append
3. If there is more, append more 
Or I am missing something?
```
result, err := c.VirtualMachineScaleSetVMsClient.List(ctx, resourceGroupName, virtualMachineScaleSetName, filter, selectParameter, expand)
	if err != nil {
		return nil, err
	}
	scaleSetsVMs = append(scaleSetsVMs, result.Values()...)
	for result.NotDone() {
		err = result.NextWithContext(ctx)
		if err != nil {
			return nil, err
		}
		scaleSetsVMs = append(scaleSetsVMs, result.Values()...)
	}
```